aten/src/ATen/native/mps/operations/Lerp.mm

```diff
@@ -10,9 +12,116 @@
 #endif
 
 namespace at::native {
+
+namespace mps {
+
+static const char* METAL_LERP_TENSOR = R"LERP_TENSOR_MTL(
+
+#include <metal_stdlib>
+using namespace metal;
+
+template<typename T>
+kernel void lerp(constant T* self       [[buffer(0)]],
+                 constant T* end        [[buffer(1)]],
+                 constant T* weight     [[buffer(2)]],
+                 device T* output       [[buffer(3)]],
+                 uint index [[thread_position_in_grid]]) {
+    output[index] = self[index] + weight[index] * (end[index] - self[index]);
+}
+
+template
+[[host_name("lerp_float")]]
+kernel void lerp<float>(constant float* self [[buffer(0)]],
+                        constant float* end  [[buffer(1)]],
+                        constant float* weight [[buffer(2)]],
+                        device float* output [[buffer(3)]],
+                        uint index [[thread_position_in_grid]]);
+
+template
+[[host_name("lerp_half")]]
+kernel void lerp<half>(constant half* self [[buffer(0)]],
+                       constant half* end  [[buffer(1)]],
+                       constant half* weight [[buffer(2)]],
```

Comment:
According to the spec, I think the weights need to be in float
https://pytorch.org/docs/stable/generated/torch.lerp.html

It could be the reason `half` test is failing.

Couple of other suggestions:
1. Do we know what happens for other dtypes, such Int and uint etc.
2. Are you handling broadcasting for start and end ?

Reply:
> It could be the reason `half` test is failing.
Updated weight to float.
 
>     1. Do we know what happens for other dtypes, such Int and uint etc.

If I am not wrong, lerp kernel tensor is not supported for int. The following snippet throws `RuntimeError: "lerp_kernel_tensor" not implemented for 'Int'`
```py
a = torch.randint(5, size=(1, 5)).int()
b = torch.randint(5, size=(1, 5)).int()
w = torch.randn(1, 5).int()
c = torch.lerp(a, b, w).sum()
```
 
>     2. Are you handling broadcasting for start and end ?
No, I am not handling broadcasting here. Should I handle it here?

The failure can be probably because lerp backward is not implemented.

Reply:
Yes, please add support for it here or add a test and exit gracefully by raising an error .

Reply:
From the docs here,
https://github.com/pytorch/pytorch/blob/2764ead4296541b324ae0e8e2b6200eec0e90c4e/aten/src/ATen/TensorIterator.h#L31-L33
I update the kernel to use `TensorIterator` as it can handle broadcasting. But the test is still failing. I am not sure what I am missing. Can you please guide me here?